hoomd/hpmc/pytest/test_external.py

```diff
@@ -0,0 +1,188 @@
+        reference_orientations = snapshot.particles.orientation
+    else:
+        reference_positions = [[0, 0, 0], [0, 0, 0]]
+        reference_orientations = [[1, 0, 0, 0], [1, 0, 0, 0]]
+    lattice = hoomd.hpmc.external.field.Harmonic(
+        reference_positions=reference_positions,
+        reference_orientations=reference_orientations,
+        k_translational=1.0,
+        k_rotational=1.0,
+        symmetries=[[1, 0, 0, 0]])
+    mc.external_potential = lattice
+
+    # create C++ mirror classes and set parameters
+    sim.run(0)
+
+    # make sure objecst are attached
+    assert mc._attached
+    assert lattice._attached
+
+
+@pytest.mark.cpu
+def test_detaching(device, simulation_factory, two_particle_snapshot_factory):
+    mc = hoomd.hpmc.integrate.Sphere()
+    mc.shape['A'] = dict(diameter=0)
+
+    # create simulation & attach objects
+    sim = simulation_factory(two_particle_snapshot_factory())
+    sim.operations.integrator = mc
+
+    # create harmonic field
+    snapshot = sim.state.get_snapshot()
+    if device.communicator.rank == 0:
+        reference_positions = snapshot.particles.position
+        reference_orientations = snapshot.particles.orientation
+    else:
+        reference_positions = [[0, 0, 0], [0, 0, 0]]
+        reference_orientations = [[1, 0, 0, 0], [1, 0, 0, 0]]
+    lattice = hoomd.hpmc.external.field.Harmonic(
+        reference_positions=reference_positions,
+        reference_orientations=reference_orientations,
+        k_translational=1.0,
+        k_rotational=1.0,
+        symmetries=[[1, 0, 0, 0]])
+    mc.external_potential = lattice
+
+    # create C++ mirror classes and set parameters
+    sim.run(0)
+
+    # make sure objecst are attached
+    sim.operations.remove(mc)
+    assert not mc._attached
+    assert not lattice._attached
+
+
+@pytest.mark.cpu
+def test_harmonic_displacement_energy(device, simulation_factory,
+                                      two_particle_snapshot_factory):
+    """Ensure harmonic displacements result in expected energy."""
+    mc = hoomd.hpmc.integrate.Sphere()
+    mc.shape['A'] = dict(diameter=0, orientable=True)
+
+    # create simulation & attach objects
+    sim = simulation_factory(two_particle_snapshot_factory())
+    sim.operations.integrator = mc
+
+    # create harmonic field
+    k_trans = 1.0
+    snapshot = sim.state.get_snapshot()
+    if device.communicator.rank == 0:
+        reference_positions = snapshot.particles.position
+        reference_orientations = snapshot.particles.orientation
+    else:
+        reference_positions = [[0, 0, 0], [0, 0, 0]]
+        reference_orientations = [[1, 0, 0, 0], [1, 0, 0, 0]]
+    lattice = hoomd.hpmc.external.field.Harmonic(
+        reference_positions=reference_positions,
+        reference_orientations=reference_orientations,
+        k_translational=k_trans,
+        k_rotational=1.0,
+        symmetries=[[1, 0, 0, 0]])
+    mc.external_potential = lattice
```

Comment:
This code appears in most tests. It would be a great canidate for a fixture that returns a simulation populated with the integrator and external potential.

Reply:
Feel free to clean this up if desired.